lms/models/group_info.py

```diff
@@ -0,0 +1,90 @@
+import sqlalchemy as sa
+
+from lms.db import BASE
+
+__all__ = ["GroupInfo"]
+
+
+class GroupInfo(BASE):
+    """
+    Some info about an LMS group that was created in h.
+
+    This info is stored purely for metrics/analytics purposes and shouldn't be
+    used for application logic. The app should treat the h API as the
+    "single-source of truth" about what h groups exist and what their IDs and
+    other properties are.
+    """
+
+    __tablename__ = "group_info"
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    #: The authority_provided_id of the group in h.
+    #:
+    #: This corresponds to the ID part of the groupid that's used in h's groups
+    #: API. For example if the groupid is "group:SOME_ID@lms.hypothesis.is"
+    #: then the authority_provided_id is the "SOME_ID" part without the leading
+    #: "group:" or the trailing "@lms.hypothes.is".
+    #:
+    #: This also corresponds to the group.authority_provided_id column in h's
+    #: DB.
+    authority_provided_id = sa.Column(sa.UnicodeText(), nullable=False, unique=True)
+
+    #: The LTI consumer_key (oauth_consumer_key) of the application instance
+    #: that this access token belongs to.
+    consumer_key = sa.Column(
+        sa.String(),
+        sa.ForeignKey("application_instances.consumer_key", ondelete="cascade"),
+        nullable=False,
+    )
+
+    #: The ApplicationInstance that this group belongs to.
+    application_instance = sa.orm.relationship(
+        "ApplicationInstance", back_populates="group_infos"
+    )
```

Comment:
Again, we're not actually using this, but this just means that `my_group_info.application_instance` is the related `ApplicationInstance` object

Reply:
Ha! On first reading of this in the tests I thought this was an instance of the `application` model. It's a kind of terrible name.